### PR TITLE
DDT-489 add booking cancellation tests

### PR DESCRIPTION
**Pull request by pimg**
resolves: DDT-489

### CODE REVIEW

File: src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+		bookingRequest.put("communicationChannel", "AO");
+		bookingRequest.put("expectedDepartureDate", "2021-05-17");
+		bookingRequest.put("isEquipmentSubstitutionAllowed", true);
+
+		Map<String, Object> commodity = new LinkedHashMap<>();
+		commodity.put("commodityType", "Bloom");
+		commodity.put("cargoGrossWeight", 2000.0);
+		commodity.put("cargoGrossWeightUnit", "LBR");
+    bookingRequest.put("commodities", Arrays.asList(commodity));
+
```
Comment (nt-as) [inline src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java:59]: Weird spacing.

File: src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	private List<Map> findAllBookings() {
+		return given()
+			.auth()
+			.oauth2(Configuration.accessToken)
+			.get(Configuration.ROOT_URI + BOOKING_SUMMARIES_PATH)
+			.then()
+			.assertThat()
+			.statusCode(200)
+			.body("size()", greaterThanOrEqualTo(0))
```
Comment (nt-as) [inline src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java:84]: Should this be `greaterThanOrEqualTo(1)` given that `testCancelBookingInInvalidDocumentStatus` will fail with `IndexOutOfBoundsException` if there are no bookings?

Comment (nt-as) [reply src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java:84]: Hah, turns out the CI fails for exactly this reason. :D

File: src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java (hunk 1 above)
Comment (pimg) [reply src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java]: fixed spacing

File: src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  }
+
+  @Test
+  public void testCancelBookingInInvalidDocumentStatus() {
+
+    given()
+        .auth()
+        .oauth2(Configuration.accessToken)
+        .when()
+        .pathParam("carrierBookingRequestReference", bookingsNotAllowedToCancel.get(0))
```
Comment (Abdiiir) [inline src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java:141]: Would it not make sense to test for all the given test-data?

**Review by Abdiiir — CHANGES_REQUESTED** [review]
specify that the relevant json-schemas are respected

**Review by Abdiiir — APPROVED** [review]

File: src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java (hunk 3 above)
Comment (pimg) [reply src/test/java/org/dcsa/api_validator/bkg/v1/cancellations/CancelBookingTest.java]: I changed the test so it executes on all bookings found with an invalid status

**Review by Abdiiir — APPROVED** [review]

**Pull request merged**